File: content/skills/agility/pyramid/src/main/java/org/jesse/game/content/skills/agility/pyramid/AgilityPyramid.java
package org.jesse.game.content.skills.agility.pyramid;

import org.jesse.game.content.skills.agility.AbstractAgilityCourse;
import org.jesse.game.content.skills.agility.AgilityCourseObstacle;
import org.jesse.game.world.entity.player.Player;
import org.jesse.game.world.entity.player.SkillConstants;
import org.jesse.game.world.entity.player.VarManager;
import org.jesse.game.world.object.WorldObject;
import org.jesse.utils.StaticInitializer;

@StaticInitializer
public final class AgilityPyramid extends AbstractAgilityCourse {

    public static final double MAX_COMPLETION_BONUS = 1000;
    public static final int HIDE_PYRAMID_VARBIT = 1556;
    public static final int MOVING_BLOCK_VARBIT = 1550;

    static {
        VarManager.appendPersistentVarbit(HIDE_PYRAMID_VARBIT);
        VarManager.appendPersistentVarbit(MOVING_BLOCK_VARBIT);
    }

    @Override
    public double getAdditionalCompletionXP() {
        return 0;
    }

    @Override
    public void runCourseObstacle(Player player, WorldObject object, AgilityCourseObstacle obstacle) {
        obstacle.updateStage(player);

        final double completionExp = Math.min(AgilityPyramid.MAX_COMPLETION_BONUS,
                300 + player.getSkills().getLevelForXp(SkillConstants.AGILITY) * 8);
        obstacle.handle(player, object, (obstacle instanceof Doorway) ? completionExp : 0, null);
    }

}

File: content/skills/agility/pyramid/src/main/kotlin/org/jesse/game/content/skills/agility/pyramid/AgilityPyramid.kt
package org.jesse.game.content.skills.agility.pyramid

import org.jesse.game.content.skills.agility.AbstractAgilityCourse
import org.jesse.game.content.skills.agility.AgilityCourseObstacle
import org.jesse.game.world.entity.player.Player
import org.jesse.game.world.entity.player.SkillConstants
import org.jesse.game.world.entity.player.VarManager
import org.jesse.game.world.`object`.WorldObject
import org.jesse.utils.StaticInitializer
import kotlin.math.min

@StaticInitializer
class AgilityPyramid : AbstractAgilityCourse() {
    override fun getAdditionalCompletionXP(): Double = 0.0

    override fun runCourseObstacle(player: Player, `object`: WorldObject?, obstacle: AgilityCourseObstacle) {
        obstacle.updateStage(player)

        val completionExp = min(
            MAX_COMPLETION_BONUS,
            (300 + player.getSkills().getLevelForXp(SkillConstants.AGILITY) * 8).toDouble()
        )
        obstacle.handle(player, `object`, if (obstacle is Doorway) completionExp else 0.0, null)
    }

    companion object {
        const val MAX_COMPLETION_BONUS: Double = 1000.0
        const val HIDE_PYRAMID_VARBIT: Int = 1556
        const val MOVING_BLOCK_VARBIT: Int = 1550

        init {
            VarManager.appendPersistentVarbit(HIDE_PYRAMID_VARBIT)
            VarManager.appendPersistentVarbit(MOVING_BLOCK_VARBIT)
        }
    }
}
